src/controller/java/src/chip/devicecontroller/ICDDeviceInfo.java

```diff
@@ -17,16 +17,129 @@
  */
 package chip.devicecontroller;
 
+import java.util.HashSet;
+import java.util.Set;
+
 /** Class for holding ICD Device information. */
 public class ICDDeviceInfo {
-  private byte[] symmetricKey;
+  /** List of methods to trigger the ICD to switch to Active mode. */
+  public enum UserActiveModeTriggerBitmap {
+    PowerCycle(0),
+    SettingsMenu(1),
+    CustomInstruction(2),
+    DeviceManual(3),
+    ActuateSensor(4),
+    ActuateSensorSeconds(5),
+    ActuateSensorTimes(6),
+    ActuateSensorLightsBlink(7),
+    ResetButton(8),
+    ResetButtonLightsBlink(9),
+    ResetButtonSeconds(10),
+    ResetButtonTimes(11),
+    SetupButton(12),
+    SetupButtonSeconds(13),
+    SetupButtonLightsBlink(14),
+    SetupButtonTimes(15),
+    AppDefinedButton(16);
+
+    private final int bitIndex;
+
+    UserActiveModeTriggerBitmap(int bitIndex) {
+      this.bitIndex = bitIndex;
+    }
+
+    public int getBitIndex() {
+      return bitIndex;
+    }
+  }
+
+  private final byte[] symmetricKey;
+  private final Set<UserActiveModeTriggerBitmap> userActiveModeTriggerHint;
+  private final String userActiveModeTriggerInstruction;
+  private final long icdNodeId;
+  private final long icdCounter;
+  private final long monitoredSubject;
 
-  public ICDDeviceInfo(byte[] symmetricKey) {
+  ICDDeviceInfo(
+      byte[] symmetricKey,
+      Set<UserActiveModeTriggerBitmap> userActiveModeTriggerHint,
+      String userActiveModeTriggerInstruction,
+      long icdNodeId,
```

Comment:
here, I prefer adding fabricId here for convenience. In addition ICDDeviceInfo is fabric-scoped. thanks

Reply:
Since ICDClientInfo contains FabricIndex (included in ScopedNodeID), I think we need to add both. I will add both FabricID and FabricIndex in the Class.

Reply:
thanks